src/java.base/macosx/classes/sun/nio/fs/BsdFileSystemProvider.java

```diff
@@ -70,6 +75,11 @@ public DynamicFileAttributeView getFileAttributeView(Path obj,
             return new BsdUserDefinedFileAttributeView(UnixPath.toUnixPath(obj),
                     Util.followLinks(options));
         }
+        if (name.equals("basic")) {
+            UnixPath file = UnixPath.toUnixPath(obj);
+            boolean followLinks = Util.followLinks(options);
+            return new BsdBasicFileAttributeView(file, followLinks);
```

Comment:
PosixFileAttributeView extends BasicFileAttributeView so it's possible to obtain the former and invoke setTimes too. I need to think more on how to make that work.

Reply:
Perhaps `setTimes()` could be moved to a static method invoked by both `BsdBasicFileAttributeView` and a new `BsdPosixFileAttributeView`?